language-plutus-core/src/Language/PlutusCore/Evaluation/Result.hs

```diff
@@ -47,9 +47,18 @@ instance Alternative EvaluationResult where
     EvaluationSuccess x <|> _ = EvaluationSuccess x
     EvaluationFailure   <|> a = a
 
+instance PrettyConst a => PrettyConst (EvaluationResult a) where
+    prettyConst (EvaluationSuccess x) = prettyConst x
+    prettyConst EvaluationFailure     = "Failure"
+
+    -- TODO: this suggests that the 'PrettyConst' class should be split into two separate classes as
+    -- this definition is pretty senseless.
+    parseConst "Failure" = EvaluationFailure
+    parseConst string    = EvaluationSuccess $ parseConst string
```

Comment:
Kind of annoying that we need this instance, but we do. See also the comment.

Reply:
I see what you mean.  I haven't thought the `parseConst` stuff through yet: maybe having them both in the same class isn't such a good idea.  It'd be a bit annoying to split them up though.  Surely every time you have a constant you'd want to implement both.  I suppose the same is true for `Read` and `Show` though.

Reply:
Actually, I'll remove `parseConst` from this PR and put it back when I get round to #1936.